runtime/services/lightd/effects.js

```diff
@@ -87,6 +147,10 @@ module.exports = LightRenderingContextManager
  */
 function LightRenderingContextManager () {
   this.id = 0
+  this.ledsConfig = light.getProfile()
+  for (var i = 0; i < this.ledsConfig.leds; ++i) {
+    realTimeRGBData.push(new Color(0, 0, 0, 1))
```

Comment:
`realTimeRGBData` could this be maintained at the instance of `LightRenderingContextManager`?

Reply:
No, every instance of  `LightRenderingContext` need to use this data. If it is defined in `LightRenderingContextManager`, there will be circular reference.